code/datums/components/storage/concrete/extract_inventory.dm

```diff
@@ -0,0 +1,33 @@
+/datum/component/storage/concrete/extract_inventory
+	max_combined_w_class = WEIGHT_CLASS_TINY * 3
+	max_items = 3
+	insert_preposition = "in"
+	attack_hand_interact = FALSE
+	quickdraw = FALSE
+	can_transfer = FALSE
+	drop_all_on_deconstruct = FALSE
+	locked = TRUE
+	rustle_sound = FALSE
+	silent = TRUE
+	var/obj/item/slimecross/reproductive/parentSlimeExtract
+//These need to be false in order for the extract's food to be unextractable
+//from the inventory
+
+/datum/component/storage/concrete/extract_inventory/Initialize()
+	. = ..()
+	set_holdable(/obj/item/food/monkeycube, list(/obj/item/food/monkeycube/syndicate, /obj/item/food/monkeycube/gorilla, /obj/item/food/monkeycube/chicken, /obj/item/food/monkeycube/bee))
+	parentSlimeExtract = parent
```

Comment:
You should be checking if parent is this type and returning COMPONENT_INCOMPATIBLE if not.